federation/pkg/kubefed/init/init.go

```diff
@@ -638,20 +640,23 @@ func createControllerManagerKubeconfigSecret(clientset client.Interface, namespa
 	return util.CreateKubeconfigSecret(clientset, config, namespace, kubeconfigName, name, "", dryRun)
 }
 
-func createPVC(clientset client.Interface, namespace, svcName, federationName, etcdPVCapacity string, dryRun bool) (*api.PersistentVolumeClaim, error) {
+func createPVC(clientset client.Interface, namespace, svcName, federationName, etcdPVCapacity, etcdPVStorageClass string, dryRun bool) (*api.PersistentVolumeClaim, error) {
 	capacity, err := resource.ParseQuantity(etcdPVCapacity)
 	if err != nil {
 		return nil, err
 	}
 
+	annotations := map[string]string{federation.FederationNameAnnotation: federationName}
+	if len(etcdPVStorageClass) > 0 {
+		annotations["volume.beta.kubernetes.io/storage-class"] = etcdPVStorageClass
```

Comment:
Is there anyway to change this behavior based on k8s cluster version?
* k8s.version >= 1.6
  * Use GA storageClass field
* k8s.version >= 1.4 && k8s.version < 1.6
  * Use beta annotation
* k8s.version >= 1.2 && k8s.version < 1.4
  * Use alpha annotation
* k8s.version < 1.2
  * Unsupported